File: examples/02-slide-layouts/main.go
// examples/02-slide-layouts/main.go demonstrates all available slide layout types.
//
// Shows how to select title-only, blank, centered-title, two-column, and
// the default title-and-content layouts using the fluent SlideContent API.
//
// Run with: go run ./examples/02-slide-layouts/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	log "github.com/djinn-soul/gopptx/pkg/stdlog"

	"github.com/djinn-soul/gopptx/pkg/pptx"
)

const (
	outputDir  = "examples/output"
	outputFile = "02_slide_layouts.pptx"
)

func main() {
	if err := run(); err != nil {
		fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}

func run() error {
	if err := os.MkdirAll(outputDir, 0o750); err != nil {
		return fmt.Errorf("create output directory: %w", err)
	}

	// Slide 1: Default title-and-content layout
	slideDefault := pptx.NewSlide("Title and Content (Default)").
		AddBullet("This is the default layout.").
		AddBullet("Used when no layout is specified.")

	// Slide 2: Title-only layout (no content placeholder)
	slideTitleOnly := pptx.NewSlide("Title Only Layout").
		WithTitleOnlyLayout()

	// Slide 3: Blank layout (no placeholders at all)
	slideBlank := pptx.NewSlide("").
		WithBlankLayout()

	// Slide 4: Centered title layout
	slideCentered := pptx.NewSlide("Centered Title Layout").
		WithCenteredTitleLayout()

	// Slide 5: Two-column layout with bullet content
	slideTwoColumn := pptx.NewSlide("Two Column Layout").
		WithTwoColumnLayout().
		AddBullet("First item in the content area.").
		AddBullet("Second item in the content area.")

	slides := []pptx.SlideContent{
		slideDefault,
		slideTitleOnly,
		slideBlank,
		slideCentered,
		slideTwoColumn,
	}

	data, err := pptx.CreateWithSlides("Slide Layout Types", slides)
	if err != nil {
		return fmt.Errorf("create presentation: %w", err)
	}

	outputPath := filepath.Join(outputDir, outputFile)
	if err = os.WriteFile(outputPath, data, 0o600); err != nil {
		return fmt.Errorf("write file: %w", err)
	}

	log.Printf("Generated %s\n", outputPath)
	return nil
}

File: examples/02-slide-layouts/main.py
"""Demonstrate all available slide layout types in gopptx.

This example demonstrates:
- Title-and-content (default) layout
- Title-only layout
- Blank layout
- Two-column layout
- Centered title layout
"""

from __future__ import annotations

from pathlib import Path

from gopptx import Presentation
from gopptx.presentation.slides import SlideLayoutType


def _add_default_slide(prs: Presentation) -> None:
    """Add a slide using the default title-and-content layout."""
    prs.add_bullet_slide(
        "Title and Content (Default)",
        [
            "This is the default layout.",
            "Used when no layout is specified.",
        ],
    )


def _add_title_only_slide(prs: Presentation) -> None:
    """Add a slide using the title-only layout."""
    prs.add_slide("Title Only Layout", layout=SlideLayoutType.TITLE_ONLY)


def _add_blank_slide(prs: Presentation) -> None:
    """Add a blank slide with no placeholders."""
    prs.add_slide("", layout=SlideLayoutType.BLANK)


def _add_two_column_slide(prs: Presentation) -> None:
    """Add a two-column layout slide with bullet content."""
    prs.add_bullet_slide(
        "Two Column Layout",
        [
            "First item in the content area.",
            "Second item in the content area.",
        ],
    )


def _add_centered_slide(prs: Presentation) -> None:
    """Add a slide with a centered title layout."""
    prs.add_slide("Centered Title Layout", layout=SlideLayoutType.CENTERED_TITLE)


def main() -> None:
    """Create a presentation demonstrating all slide layout types."""
    output_dir = Path("examples/output")
    output_dir.mkdir(exist_ok=True)

    with Presentation.new("Slide Layout Types") as prs:
        _add_default_slide(prs)
        _add_title_only_slide(prs)
        _add_blank_slide(prs)
        _add_two_column_slide(prs)
        _add_centered_slide(prs)

        output_path = output_dir / "02-slide-layouts.pptx"
        prs.save(str(output_path))
        print(f"Saved: {output_path}")

    print("\n=== SUMMARY ===")
    print("Created 5 slides, each with a different layout type:")
    print("  - Default (title + content)")
    print("  - Title Only")
    print("  - Blank")
    print("  - Two Column")
    print("  - Centered Title")


if __name__ == "__main__":
    main()
